ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                                 int bUpdateIn, CSLConstList papszOpenOptions,
+                                 struct MiraMonVectMapInfo *MMMap)
+    : poFeatureDefn(nullptr), iNextFID(0), phMiraMonLayer(nullptr),
+      hMiraMonLayerPNT(), hMiraMonLayerARC(), hMiraMonLayerPOL(),
+      hMiraMonLayerReadOrNonGeom(), hMMFeature(),
+      bUpdate(CPL_TO_BOOL(bUpdateIn)), nMMMemoryRatio(1.0),
+      m_fp(fp ? fp : VSIFOpenL(pszFilename, (bUpdateIn ? "r+" : "r"))),
+      papszValues(nullptr), padfValues(nullptr), bValidFile(false)
+{
+
+    CPLDebug("MiraMon", "Creating/Opening MiraMon layer...");
+    /* -------------------------------------------------------------------- */
+    /*      Create the feature definition                                   */
+    /* -------------------------------------------------------------------- */
+    poFeatureDefn = new OGRFeatureDefn(CPLGetBasename(pszFilename));
+    SetDescription(poFeatureDefn->GetName());
+    poFeatureDefn->Reference();
+
+    if (bUpdate)
+    {
+        /* ---------------------------------------------------------------- */
+        /*      Establish the version to use                                */
+        /* ---------------------------------------------------------------- */
+        const char *pszVersion = CSLFetchNameValue(papszOpenOptions, "Version");
+        int nMMVersion;
+
+        if (pszVersion)
+        {
+            if (EQUAL(pszVersion, "V1.1"))
+                nMMVersion = MM_32BITS_VERSION;
+            else if (EQUAL(pszVersion, "V2.0") ||
+                     EQUAL(pszVersion, "last_version"))
+                nMMVersion = MM_64BITS_VERSION;
+            else
+                nMMVersion = MM_32BITS_VERSION;  // Default
+        }
+        else
+            nMMVersion = MM_32BITS_VERSION;  // Default
+
+        /* ---------------------------------------------------------------- */
+        /*      Establish the charset of the .dbf files                     */
+        /* ---------------------------------------------------------------- */
+        const char *pszdbfEncoding =
+            CSLFetchNameValue(papszOpenOptions, "DBFEncoding");
+        char nMMRecode;
+
+        if (pszdbfEncoding)
+        {
+            if (EQUAL(pszdbfEncoding, "UTF8"))
+                nMMRecode = MM_RECODE_UTF8;
+            else  //if (EQUAL(pszdbfEncoding, "ANSI"))
+                nMMRecode = MM_RECODE_ANSI;
+        }
+        else
+            nMMRecode = MM_RECODE_ANSI;  // Default
+
+        /* ----------------------------------------------------------------- */
+        /*      Establish the nMemoryRatio to use                            */
+        /* ----------------------------------------------------------------- */
+        const char *pszMemoryRatio =
+            CSLFetchNameValue(papszOpenOptions, "CreationMemoryRatio");
+
+        if (pszMemoryRatio)
+            nMMMemoryRatio = atof(pszMemoryRatio);
```

Comment:
you probably want to clamp the user input to avoid crazy inputs. Not sure if [0.5, 2] is the appropriate interval though
```suggestion
            nMMMemoryRatio = std::clamp(CPLAtof(pszMemoryRatio), 0.5, 2);
```

Reply:
It could be any value greater than 0.5 (sounds good) but no limit (if the machine has lots of RAM). So, recommended, 1, 2, 6... 8... 10... but perhaps 1000 it's too much. Its a value to make some buffers biggers to make the driver less accesses to disk.
So, it can be clamped like that? no superior limit? Or 1000, perhaps... from 0.5 to 1000, could be a good aproximation.

Reply:
Your call about an appropriate upper bound, but honestly I would prefer this option to not be exposed to users at all. I suspect 99.9% of them will struggle understand what this does and which value to put. If a user specifies a too large value, isn't there a risk that the driver tries to allocate more RAM than available and blow up?

Reply:
I think it's a necessary value for big layers but it could be hidden from doc, if you want, but I'm sure my boss want it. So, a limitation is a good thing to avoid blow up anything.
FFor instance, a 1.000.000 polygons layers needs 0.2 Gb to work. If this value is 2, it only needs 0.206. If the value is 100, it needs 0.95. If it's 1000 it needs 7.87.
So, perhaps mantain the value between 0.5 and 100 could be a good aproximation.
I hope that if one reads this value and don't understand the value, this is not going to be used and the defect value is 1. So no problems with that.
It's ok for you?  nMMMemoryRatio = std::clamp(CPLAtof(pszMemoryRatio), 0.5, 100);  ?

Reply:
> nMMMemoryRatio = std::clamp(CPLAtof(pszMemoryRatio), 0.5, 100); ?

yes...

Reply:
Ok, thanks :)
Finally it need two casts in the numbers:
nMMMemoryRatio = std::clamp(CPLAtof(pszMemoryRatio), (double)0.5f, (double)100);

Reply:
Just do:
nMMMemoryRatio = std::clamp(CPLAtof(pszMemoryRatio), 0.5, 100.0);
to have all values as double without needing any cast

Reply:
True, I forgot the .0 on the second number